### ISSUE TITLE
GitHub/Path and GitHub/Filename are not saved locally (timimi-Extension)

### ISSUE THREAD

**Issue by siramon** [Jermolene/TiddlyWiki5#4666]
Description:
When saving the tiddlywiki via the Button "Save changes", the two mentioned fields are not save in the local html-Document, so next time I open the html-File, the GitHub-Call (when saving) is not successful.

Setup (everything is up-to-date):
- Firefox
- timimi-Extension
- configured GitHub Saver

Effect:
- local html-file misses
```html
<div created="20200518154203657" modified="20200518154205588" title="$:/GitHub/Path">
<pre>/wiki/</pre>
</div>
```
and
```html
<div created="20200518154207119" modified="20200518154212372" title="$:/GitHub/Filename">
<pre>00_sh-tiddlywiki.html</pre>
</div>
```
- in the version saved to github, the two fields are saved

I can reproduce this in my setup. Let me know, if you need anything more to reproduce this effect. Also please let me know, if this is a timimi-problem. In that case i will post the bug there.

Thank you for this great App!

**Comment by siramon** [comment]
Trying to fiddle around, to get more information - question at the end.
 
The setup mentioned above was on a Windows7 machine. When hitting the save-button, the wiki was sent to github (via Github-Saver) and store locally via timimi-extension.

Know on macOS with a current firefox extension, i can't run both saving methods together. When I use Github-Saver, timimi is ignored. As soon, as i disable Github-Saver, the timimi-extension saves it correctly (Message ID is tiddlywiki-save-file-response-1).

_Question: Is it possible to use the two saving-methods together?_

**Comment by pmario** [comment]
>Question: Is it possible to use the two saving-methods together?

No. TW uses so-called "savers". Each saver has a priority and the first one that can save wins. 

IMO timimi uses the tiddlyfox saver priority which is 1500. 
github has 2000 

So timimi should win.

**Comment by siramon** [comment]
@pmario Thank you for your answer.

So the github-scenario would be like this.

Start:

1. create repository
2. add tiddlywiki.html / commit / push to github
3. open the .html from local git in the browser
3. change "saver" to github-saver / configure
4. when saving now, changes will be pushed to github (as long as the browser window stays open)

now everytime after closing the browser window:

1. update the local repository (as the .html was not changed because of "saver" priorities: pull from github)
2. open the .html from local git in the browser
3. when saving now, changes will be pushed to github (as long as the browser window stays open)

Feedback welcomed!

**Comment by Guillawme** [comment]
Looks like the same thing happens with TiddlyDesktop and the GitLab saver: the username, target repository, path to target file and target file name fields are lost after I close and reopen my wiki file. Strangely, the access token field is not lost, so upon reopening the file I only need to re-enter the other fields. It would be convenient if all fields were saved properly.

It is also not clear to me what you mean @pmario by:

> Each saver has a priority and the first one that can save wins.

Looks like in my case, both the file saved locally by TiddlyDesktop and the one in the GitLab repository have the latest changes (except the local file not keeping the fields mentioned above in the saver configuration).

**Comment by pmario** [comment]
@Guillawme ... I don't know TiddlyDesktop well. So @Jermolene ??

**Comment by Guillawme** [comment]
Actually I take it back, I just noticed inconsistencies in my wiki file. Something I am sure I added recently was not here when I reopened my wiki file saved locally (or so I thought) with TiddlyDesktop.

If at all possible, it would be really convenient if two savers could work in succession. That would allow one to have a local copy up-to-date, and a copy saved to GitHub or GitLab providing both a remote backup and a version history.

**Comment by pmario** [comment]
>If at all possible, it would be really convenient if two savers could work in succession. That would allow one to have a local copy up-to-date, and a copy saved to GitHub or GitLab providing both a remote backup and a version history.

Absolutely true. .. There is some discussion going on here. see: https://github.com/Jermolene/TiddlyWiki5/pull/4716

**Comment by siramon** [comment]
Would be awesome!

Thank you for pointing to the discussion @pmario
